src/Groups/group_characters.jl

```diff
@@ -232,14 +233,54 @@ function Base.iterate(wi::WordsIterator, state::Int)
 end
 
 
-# Utility:
-# Turn integer values to strings, but replace `0` by `"."`,
-# print irrationalities via the `Hecke.math_html` method for `nf_elem`.
-# If `alphabet` is nonempty then represent irrationalities by names
-# generated by iterating over `WordsIterator(alphabet)`,
-# and store the meanings of these strings in the form of pairs `value => name`
-# in the legend array that is returned.
-function matrix_of_strings(tbl::GAPGroupCharacterTable; alphabet::String = "")
+@doc Markdown.doc"""
+    as_sum_of_roots(val::nf_elem, root::String)
+
+Return a string representing the element `val` of a cyclotomic field
+as a sum of multiples of powers of the primitive root which is printed as
+`root`.
+"""
+function as_sum_of_roots(val::nf_elem, root::String)
+    F = parent(val)
+    flag, N = Hecke.iscyclotomic_type(F)
+    flag || error("$val is not an element of a cyclotomic field")
+
+    # `string` yields an expression of the right structure,
+    # but not in terms of `root`
+    # and without curly brackets for subscripts and superscripts.
+    str = string(val)
+    str = replace(str, "*" => "")
+    str = replace(str, string(F.S) => root*"_{"*string(N)*"}")
```

Comment:
Does this work, too? It'd be more readable, IMHO
```suggestion
    str = replace(str, string(F.S) => "$root_{$N}")
```

Reply:
No, one gets the error message that `root_` is not defined.
One could write `"$("$root")_{$N}"` but is it more readable?

Reply:
`$(root)_{$N}`.

Reply:
Yes, `"$(root)_{$N}"` works. Thanks.